File: docs/guides/javascript/src/pushSetup.ts
import type { PushTaskRecords } from 'algoliasearch';
import fs from 'node:fs';

import { algoliasearch } from 'algoliasearch';

// use the region matching your applicationID
const client = algoliasearch('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY').initIngestion({ region: 'us' });

try {
  // read local JSON file containing array of records
  const records = JSON.parse(fs.readFileSync('records.json', 'utf8')) as PushTaskRecords[];

  // setting `watch` to `true` will make the call synchronous
  const resp = await client.pushTask({
    taskID: 'YOUR_TASK_ID',
    pushTaskPayload: { action: 'addObject', records: records },
    watch: true,
  });

  console.log(resp);
} catch (err) {
  console.error(err);
}

File: docs/guides/go/src/pushSetup.go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"os"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/next/ingestion"
	"github.com/algolia/algoliasearch-client-go/v4/algolia/utils"
)

func push() {
	// use the region matching your applicationID
	client, err := ingestion.NewClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", ingestion.US)
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		panic(err)
	}

	content, err := os.ReadFile("records.json")
	if err != nil {
		panic(err)
	}

	var records []ingestion.PushTaskRecords

	err = json.Unmarshal(content, &records)
	if err != nil {
		panic(err)
	}

	// setting `watch` to `true` will make the call synchronous
	resp, err := client.PushTask(context.Background(), "YOUR_TASK_ID", ingestion.ACTION_ADD_OBJECT, records, utils.ToPtr(true))
	if err != nil {
		panic(err)
	}

	fmt.Printf("%#v\n", resp)
}
